### PR TITLE
fixed a bunch of bugs and added equivilant courses

### PR DESCRIPTION
**Pull request by Peedee2002**

### CODE REVIEW

File: backend/server/manual_fixes.py [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
 
 
-def remove_course(structure: dict, course: str) -> dict[str, dict] | None:
+def remove_course(structure: dict[str, StructureContainer], course: str):
     """indescriminantly remove a course from structure"""
     for item, collection in structure.items():
         if item == "Rules":
             continue
-        for container in collection:
+        for container in collection["content"]:
```
Comment (Peedee2002) [inline backend/server/manual_fixes.py:11]: what no typing does to a mf

Comment (imagine-hussain) [reply backend/server/manual_fixes.py:11]: REAL!

File: backend/algorithms/objects/conditions.py [hunk 2]
```diff
@@ -342,7 +342,7 @@ def __init__(self, exclusion: str):
         self.exclusion = exclusion
 
     def validate(self, user: User) -> tuple[bool, list[str]]:
-        is_valid = not user.has_taken_course(self.exclusion)
+        is_valid = not user.has_taken_specific_course(self.exclusion)
```
Comment (imagine-hussain) [inline backend/algorithms/objects/conditions.py:345]: 🤨

File: backend/server/manual_fixes.py [hunk 3]
```diff
@@ -1,22 +1,22 @@
+# pylint: disable=missing-module-docstring,missing-function-docstring
 from contextlib import suppress
+from server.routers.model import StructureContainer
 
 
-def remove_course(structure: dict, course: str) -> dict[str, dict] | None:
+def remove_course(structure: dict[str, StructureContainer], course: str):
     """indescriminantly remove a course from structure"""
```
Comment (imagine-hussain) [inline backend/server/manual_fixes.py:7]: I was going to comment on the spelling here but in doing so, I failed to spell indiscriminately wrong more times than I want to admit. To be honest,the "dictionary spelling" of the word is wrong, I don't know who came up with it but, I wish them a life of pain

**Review by imagine-hussain — APPROVED** [review]

File: backend/server/manual_fixes.py (hunk 3 above)
Comment (Peedee2002) [reply backend/server/manual_fixes.py:7]: yea, i will commit this mistake to history now. Whoever has no sin, cast the first stone 😆

File: backend/algorithms/objects/conditions.py (hunk 2 above)
Comment (Peedee2002) [reply backend/algorithms/objects/conditions.py:345]: yea, i know, wierd. Exclusions probably should be exhaustive, so i split off the two types.

**Pull request merged**